README.md

```diff
@@ -138,7 +138,8 @@ Optional<JSONObject> payload = appEncryptionJson.load(persistenceKey, dataPersis
 * [Key Management Service](docs/KeyManagementService.md)
 * [Metastore Persistence](docs/Metastore.md)
 * [Key Caching](docs/KeyCaching.md)
-* [Common APIs and Algorithm Internals](docs/Internals.md)
+* [Directory Structure and Common APIs](docs/DirectoryStructureAndCommonAPIs.md)
```

Comment:
given our refactoring of this content, i think it would be cleaner to simplify/generalize this to `Code Structure` or similar (ditto on filename and main header). also avoids coupling to the headers in case we add/edit more later